google/auth/crypt/base.py

```diff
@@ -62,3 +68,51 @@ def sign(self, message):
         # pylint: disable=missing-raises-doc,redundant-returns-doc
         # (pylint doesn't recognize that this is abstract)
         raise NotImplementedError('Sign must be implemented')
+
+
+class _FromServiceAccountMixin(object):
```

Comment:
@jonparrott I wanted to re-use these factories, as they truly only depend on `from_string`, but I wasn't sure how to go about it (i.e. the mixin is a little gross). WDYT?